extensions/applicationinsights-cfgsync-js/src/CfgSyncPlugin.ts

```diff
@@ -0,0 +1,412 @@
+                        };
+                        const request = new Request(url, init);
+                        doAwaitResponse(fetch(request), (result) => {
+                            if (!result.rejected) {
+                                let response = result.value;
+                                if (response.ok) {
+                                    doAwaitResponse(response.text(), (res) => {
+                                        _doOnComplete(oncomplete, response.status, res.value, isAutoSync);
+                                    });
+                                }
+                            }
+                        });
+                    } catch (e) {
+                        // eslint-disable-next-line no-empty
+                    }
+                }
+            }
+
+            function _xhrSender(url: string, oncomplete: OnCompleteCallback, isAutoSync?: boolean) {
+                try {
+                    let xhr = new XMLHttpRequest();
+                    xhr.open(STR_GET_METHOD, url);
+                    xhr.onreadystatechange = () => {
+                        if (xhr.readyState === XMLHttpRequest.DONE) {
+                            _doOnComplete(oncomplete, xhr.status, xhr.responseText, isAutoSync);
+                        }
+                    }
+                    xhr.send();
+                } catch (e) {
+                    // eslint-disable-next-line no-empty
+                }
+            }
+
+            function _onFetchComplete(status: number, response?: string, isAutoSync?: boolean) {
+                try {
+                    if (status >= 200 && status < 400 && response) {
+                        _retryCnt = 0; // any successful response will reset retry count to 0
+                        let JSON = getJSON();
+                        if (JSON) {
+                            let cfg = JSON.parse(response); //comments are not allowed
+                            cfg && _setCfg(cfg, isAutoSync);
+                        }
+                    } else {
+                        _retryCnt ++;
+                    }
+                } catch (e) {
+                    // eslint-disable-next-line no-empty
+                }
+            }
+
+            function _doOnComplete(oncomplete: OnCompleteCallback, status: number, response?: string, isAutoSync?: boolean) {
+                try {
+                    oncomplete(status, response, isAutoSync);
+                } catch (e) {
+                    // eslint-disable-next-line no-empty
+                }
+            }
+
+            function _addEventListener() {
+                if (_receiveChanges) {
+                    let global = getGlobal();
+                    if (global) {
+                        try {
+                            eventOn(global, _evtName, (event: any) => {
+                                let cfgEvent = event && (event as any).detail;
+                                if (_onCfgChangeReceive && cfgEvent) {
```

Comment:
For later: We should probably add some simple validation so that (if) there is a clash with the event name we don't start crashing the page -- ie. something simple like checking that its a plain Object (`isPlainObject()`) and maybe in the future we include a "name" or somethign like that for us th be able to check